js-core/lesson3.5.md

```diff
@@ -0,0 +1,109 @@
+![](https://img.shields.io/badge/status-draft-darkred.svg) 
+
+
+# Pair Programming Day
+** What we will learn today?**
+- We will practice pair programming with a mentor
+---
+
+## How to work today?
+Today, we want to focus on our problem solving skills, how we think and tackle problems before even code, then crafting a **beautiful** solution using techniques like **TDD**
+- Think about the solution before writing any code (maybe even write pseudocode, or pencil a solution)
+- Break bigger tasks into smaller pieces, and solve them one step at a time.
+- Think about your function signature
+- Write Test -> Write Code Implementation -> Refactor (repeat)
+- Use the debugger
+- Good functions
+    - have good names and clear parameter names
+    - have single responsibility
+    - be defensive: don't trust your parameter
+- Good tests
+    - have good test description
+    - test for edge cases and unexpected behavior
+
+> Students should drive this exercise - make use of having a mentor with you, take a step back and let them explain anything you don't understand
+
+## The exercise
+The class will be based on Find Alive people exercise on [JS-TDD-Exercises](https://github.com/CodeYourFuture/js-tdd-exercises) repo
+
+- Implement the basic example if you haven't (find alive people)
+if you did, let the mentor review it
+
+- Write a function findByAge that finds people over a certain age (age is a parameter)
+    - signature: findByAge(age) - age is a number
+    - example: findByAge(60) returns every person over 60
+- **Refactor** findByAge to take a second paramter (boolean) that makes look for people below a certain age (rather than above)
```

Comment:
Remember to change "second" to "third" here if you agree with my previous comment.

Also there's a typo in "paramter".

Also: "that looks for people below a certain age" (without "makes")